lighthouse-core/gather/gather-runner.js

```diff
@@ -536,7 +537,24 @@ class GatherRunner {
   static async getWebAppManifest(passContext) {
     const response = await passContext.driver.getAppManifest();
     if (!response) return null;
-    return manifestParser(response.data, response.url, passContext.url);
+    const manifest = manifestParser(response.data, response.url, passContext.url);
+
+    // Add a warning for icons that aren't fetchable.
+    if (manifest.value && Array.isArray(manifest.value.icons.value)) {
+      const iconsToFetch = manifest.value.icons.value
+        .filter(icon => icon.value.src && icon.value.src.value);
```

Comment:
were empty string icons treated as acceptable before but now they won't be? or I suppose they will be valid because we won't try to fetch and put any warning on them?

Reply:
at this point empty strings were converted to `undefined` (see parser). later, the icons check will ignore icons whose `src` does not end in `.png`.

... however I've just noticed there's a bug if `typeHint === 'image/png'` and `src` is undefined. this is so dirty. Can fix by tweaking `getFetchableIcons` to also check for a nonempty string.

Reply:
aight sgtm